tx/blob_tx.go

```diff
@@ -59,7 +59,7 @@ func MarshalBlobTx(tx []byte, blobs ...*share.Blob) ([]byte, error) {
 	}
 	// nil check
 	for i, b := range blobs {
-		if b == nil {
+		if b == nil || b.IsEmpty() {
```

Comment:
[nit] The name `IsEmpty` does not immediately suggest that it checks the emptiness of the namespace. I suggest either renaming the method or adding a clarifying comment here to avoid confusion. Otherwise, one might intuitively think that it checks the emptiness of blobs rather than the namespace.

Reply:
It's not about checking the emptiness of the namespace but of the entire blob. It checks the namespace as proxy for the entire blob. By construction you can't have a blob with empty data and a non-empty namespace. I could have checked that all the contents of the blob had their "null" value but only one should suffice

Reply:
> By construction you can't have a blob with empty data and a non-empty namespace.

That should be documented on the Blob method. Or you could still do the nil check for readability, even though its not possible.

Reply:
Do you want me to also add a check that blob.data is not length 0?

Reply:
Zero data check would be clearer for the code readability than cognitive overhead of understanding why its only a namespace check